hydroshare_on_jupyter/server.py

```diff
@@ -347,15 +350,28 @@ def _destroy_session(self):
 class ListUserHydroShareResources(HeadersMixIn, BaseRequestHandler):
     """List the HydroShare resources a user has edit permission of."""
 
+    def map_resource_model_types(self, res_preview_models: list[ResourcePreview]) -> List[ResourceMetadata]:
+        # take an instance of ResourcePreview (hsclient) and convert to ResourceMetadata
+        resource_metadata_models = []
+        for res_prev_model in res_preview_models:
+            data = res_prev_model.model_dump()
+            # remove hsclient-specific fields
+            for field in ("abstract", "doi", "public", "discoverable" "shareable", "coverages", "published",
+                          "resource_map_url", "science_metadata_url"):
+                data.pop(field, None)
+            resource_metadata_models.append(ResourceMetadata(**data))
+        return resource_metadata_models
+
     _custom_headers = [("Access-Control-Allow-Methods", "GET")]
 
     def get(self):
         session = self.get_session()
 
         resources = list(session.session.search(edit_permission=True))
 
-        # Marshall hsclient representation into CollectionOfResourceMetadata
-        self.write(CollectionOfResourceMetadata.parse_obj(resources).json())
+        # create CollectionOfResourceMetadata from the list of ResourceMetadata models
+        resources = self.map_resource_model_types(resources)
+        self.write(CollectionOfResourceMetadata.model_validate(resources).model_dump_json())
```

Comment:
@pkdash can you explain what we are doing here?

Reply:
The changes in line 374 are due to changes from pydantic v1 to v2. The new line that I added is line 373. In pydantic v1, it was possible to take a list of ResourcePreview models/objects (obtained using hsclient) and parse those into ResourceMetadata objects as the CollectionOfResourceMetadata. That auto mapping from ResourcePreview to ResourceMetadata did not work in pydantic v2 (due to stricter validation rules). That's why I had to do a custom mapping in line 373. Line 374 is a json response to the frontend providing a list of resources. Hope I answered your question.

Reply:
thanks for the explanation!